nncf/quantization/algorithms/accuracy_control/ranker.py

```diff
@@ -0,0 +1,301 @@
+
+        :param groups_to_rank: Groups of quantizers that should be ranked.
+        :param initial_model: Initial not quantized model.
+        :param quantized_model: Quantized model.
+        :param quantized_model_graph: NNCF graph for quantized model.
+        :param excluded_groups: Groups that should be excluded from ranking.
+        :return: List of ranked groups of quantizers.
+        """
+        # See `Ranker.__init__()` to understand why we should do this.
+        if self._ref_values is None:
+            self._ref_values = self._collect_values_for_each_item(initial_model,
+                                                                  self._get_data_items())
+
+        approx_values = self._collect_values_for_each_item(quantized_model,
+                                                           self._get_data_items())
+
+        # Create a subset of data items that will be used to rank groups of quantizers.
+        scores = [
+            self._ranking_fn(ref_val, approx_val) for ref_val, approx_val in zip(self._ref_values, approx_values)
+        ]
+        ranking_subset_indices = get_ranking_subset_indices(scores, self._ranking_subset_size)
+        # TODO(andrey-churkin): The ranking subset size usually is small. So it is possible
+        # to save all ranking data items in memory and don't read them again.
+        ranking_data_items = self._get_data_items(ranking_subset_indices)
+
+        # Calculate ranking score for groups of quantizers.
+        ranking_scores = []  # ranking_scores[i] is the ranking score for groups_to_rank[i]
+        for current_group in groups_to_rank:
+            if excluded_groups and current_group in excluded_groups:
+                continue
+
+            modified_model = revert_operations_to_floating_point_precision(
+                current_group.operations,
+                current_group.quantizers,
+                quantized_model,
+                quantized_model_graph,
+                self._algo_backend.create_command_to_remove_quantizer,
+                self._algo_backend.create_command_to_update_bias,
+                self._algo_backend.create_command_to_update_weight)
+
+            # Calculate the ranking score for the current group of quantizers.
+            ranking_score = self._calculate_ranking_score(modified_model, ranking_data_items, ranking_subset_indices)
+
+            # TODO:
+            ranking_scores.append(float(ranking_score))
+
+        # Rank groups.
+        ranked_groups = [
+            group for _, group in sorted(zip(ranking_scores, groups_to_rank), key=operator.itemgetter(0))
+        ]
+
+        return ranked_groups
+
+    @abstractmethod
+    def _get_data_items(self, indices: Optional[List[int]] = None) -> Iterable[Any]:
+        """
+        :param indices:
+        :return:
+        """
+
+    @abstractmethod
+    def _collect_values_for_each_item(self, model, data_items: Iterable[Any]) -> List[Any]:
```

Comment:
The type hint for the model was missed here and below. Also, the methods descriptions are empty.

Reply:
Done